metricshub-agent/src/main/java/org/metricshub/cli/SnmpV3Cli.java

```diff
@@ -378,6 +389,13 @@ public Integer call() throws Exception {
 					displayQuery(queryNode.get("action").asText(), queryNode.get("oid").asText());
 					// Execute the SNMPv3 query
 					final String result = extension.executeQuery(configuration, queryNode);
+					// Save the snmp result to a file if the filename is provided when the record option is enabled
+					// CHECKSTYLE:OFF
+					if ("WALK".equalsIgnoreCase(queryNode.get("action").asText()) && record) {

```

Comment:
The code is not the same as the one developed for `SnmpCli`.

Reply:
`queryNode.get("action").asText()` should be created as a variable and the if check should be as follows:
`if ("walk".equals(action) && record) {`